bouncer/src/repo/manipulator/modeloptimizer/repo_optimizer_multipart.h

```diff
@@ -48,48 +48,82 @@ namespace repo {
 				*/
 				bool apply(repo::core::model::RepoScene *scene);
 
+				/*
+				* Maps from a material UUID in the original scene graph, and the duplicate node in the stash graph
+				*/
+				typedef std::unordered_map<repo::lib::RepoUUID, repo::lib::RepoUUID, repo::lib::RepoUUIDHasher> MaterialUUIDMap;
+
+				/*
+				* A dictionary of MaterialNodes for a set of UUIDs
+				*/
+				typedef std::unordered_map<repo::lib::RepoUUID, repo::core::model::RepoNode*, repo::lib::RepoUUIDHasher> MaterialNodes;
```

Comment:
I have updated it to use MaterialNode! 

For the name, it actually is kind of. It is used to accumulate the results from `processSupermeshMaterials`, where the actual stash nodes are created on demand (before then, only placeholder uuids are created). So, for a while they only exist in this structure. Do you think MaterialMap is still the better name?

Reply:
yea because what you're describing is the usecase.

Whereas this is an alias for the structure, the structure is that it's a UUID map to a material

If we just call it MaterialNodes it's less clear, as it could easily be a set or vector. Naming the variable you instantiate to be materialNodes is fine though!